modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionServiceImpl.java

```diff
@@ -95,6 +95,23 @@ public CPSpecificationOption getCPSpecificationOption(
 			cpSpecificationOptionId);
 	}
 
+	@Override
+	public CPSpecificationOption getCPSpecificationOption(
+			long companyId, String key)
+		throws PortalException {
+
+		CPSpecificationOption cpSpecificationOption =
+			cpSpecificationOptionLocalService.getCPSpecificationOption(
+				companyId, key);
+
+		if (cpSpecificationOption != null) {
+			_cpSpecificationOptionModelResourcePermission.check(
+				getPermissionChecker(), cpSpecificationOption, ActionKeys.VIEW);
+		}
```

Comment:
@andrea-ale-sbarra I don't think we need the null check here. Wont `cpSpecificationOptionLocalService.getCPSpecificationOption(companyId, key);`
just throw `NoSuchCPSpecificationOptionException` if it does not find a cpSpecificaionOption?

Reply:
you're totally right! thanks for the catch @dsitu